src/main/autoUpdater.js

```diff
@@ -0,0 +1,101 @@
+const {app, BrowserWindow, dialog, ipcMain, shell} = require('electron');
+const path = require('path');
+const {autoUpdater} = require('electron-updater');
+const semver = require('semver');
+
+const interval48hours = 172800000; // 48 * 60 * 60 * 1000
+
+let updaterWindow = null;
+
+function setEvent(win, eventName) {
+  ipcMain.on(eventName, (event) => {
+    if (event.sender === win.webContents) {
+      win.emit(eventName);
+    }
+  });
+}
+
+function createUpdaterWindow(options) {
+  const windowWidth = 480;
+  const windowHeight = 240;
+  const windowOptions = {
+    title: `${app.getName()} Updater`,
+    maximizable: false,
+    show: false,
+    width: windowWidth,
+    height: windowHeight,
+    resizable: false,
+    autoHideMenuBar: true
+  };
+  if (process.platform === 'linux') {
+    windowOptions.icon = options.linuxAppIcon;
+  }
+
+  const win = new BrowserWindow(windowOptions);
+  win.once('ready-to-show', () => {
+    win.show();
+  });
+  const updaterURL = (global.isDev ? 'http://localhost:8080' : `file://${app.getAppPath()}`) + '/browser/updater.html';
+  win.loadURL(updaterURL);
+
+  setEvent(win, 'click-release-notes');
+  setEvent(win, 'click-skip');
+  setEvent(win, 'click-remind');
+  setEvent(win, 'click-install');
+
+  return win;
+}
+
+function isUpdateApplicable(appState, updateInfo) {
+  if (appState.updateCheckedDate.value - (new Date(updateInfo.releaseDate)).value < interval48hours) {
```

Comment:
Are these two dates always in a fixed time zone? Like are they both UTC dates? If either is based on the user's timezone, then we may not actually wait 48 hours.

Reply:
UTC is always used.

- updateCheckedDate: [Date.prototype.toISOString()](https://github.com/mattermost/desktop/pull/582/files/972a233ed49bceea2d0ed419c4aa0cafd60adb7f#diff-f86667a718f2bfb29e313e6c1f73a02eR13)
- releaseDate: I can't show in source code. But I confirmed that the date uses UTC ('Z' in ISO format)